sycl/CMakeLists.txt

```diff
@@ -95,34 +108,108 @@ add_custom_target(sycl-headers ALL
 COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/CL ${dst_dir}/CL
 COMMENT "Copying SYCL headers ...")
 
-# Configure SYCL headers
-install(DIRECTORY "${sycl_inc_dir}/." DESTINATION "${LLVM_INST_INC_DIRECTORY}" COMPONENT sycl-headers)
-
-# Configure OpenCL header and ICD loader
-include_directories(AFTER "${sycl_inc_dir}" "${OpenCL_INCLUDE_DIRS}")
-link_libraries(${OpenCL_LIBRARIES})
-
 # SYCL runtime library
 add_subdirectory(source)
 
 # SYCL toolchain builds all components: compiler, libraries, headers, etc.
+set(SYCL_TOOLCHAIN_DEPS
+  sycl
+  clang
+  clang-offload-wrapper
+  clang-offload-bundler
+  llc
+  llvm-as
+  llvm-ar
+  llvm-dis
+  llvm-no-spir-kernel
+  llvm-spirv
+  llvm-link
+  llvm-objcopy
+  opt)
 add_custom_target( sycl-toolchain
-  DEPENDS sycl
-          clang
-          clang-offload-wrapper
-          clang-offload-bundler
-          llc
-          llvm-as
-          llvm-ar
-          llvm-dis
-          llvm-no-spir-kernel
-          llvm-spirv
-          llvm-link
-          llvm-objcopy
-          opt
+  DEPENDS ${SYCL_TOOLCHAIN_DEPS}
   COMMENT "Building SYCL compiler toolchain..."
 )
 
 add_subdirectory( test )
 add_subdirectory( unittests )
 add_subdirectory( tools )
+
+# Configure SYCL toolchain installation targets
+add_custom_target(deploy-sycl-toolchain)
+
+set(SYCL_BUNDLE_DEPS ${SYCL_TOOLCHAIN_DEPS} opencl clang-resource-headers)
+
+foreach(comp ${SYCL_BUNDLE_DEPS})
+    add_custom_command(TARGET deploy-sycl-toolchain PRE_BUILD
+        COMMAND ${CMAKE_COMMAND}
+        ARGS -DCOMPONENT="${comp}" -P "${CMAKE_BINARY_DIR}/cmake_install.cmake")
+endforeach()
+
+set(LLVM_LIT_EXT "")
+if(WIN32)
+  set(LLVM_LIT_EXT ".py")
+endif()
+add_custom_command(TARGET deploy-sycl-toolchain PRE_BUILD
+  COMMAND ${CMAKE_COMMAND}
+  ARGS -E copy ${CMAKE_BINARY_DIR}/bin/llvm-lit${LLVM_LIT_EXT}
+  ${CMAKE_INSTALL_PREFIX}/bin/llvm-lit${LLVM_LIT_EXT})
+
+# For some reason clang-cl is being created on Linux. Remove it so users
```

Comment:
I'd suggest fix this on clang side rather than add such a post build adjustment.
Is it reproduced consistently? Do you have steps to reproduce?

Reply:
@valerydmit when I build sycl-toolchain target, I get clang-cl in bin directory. And at the time I'm not sure it is a bug in clang. There are tests that call clang-cl, and they seem to run and pass on Linux. On the other hand, all clang-cl can do on Linux is show help and crash, complaining about missing MSVC.